doc/programming_guide/tutorials/memory_usage.rst

```diff
@@ -0,0 +1,86 @@
+############
+Memory Usage
+############
+
```

Comment:
I would add somthing like: this section explores how to check the memory usage for a given application. User can get this information at compile time or after compilation.

Reply:
Added